src/api/libcellml/variable.h

```diff
@@ -429,6 +429,25 @@ class LIBCELLML_EXPORT Variable: public NamedEntity
      */
     bool hasInterfaceType(InterfaceType interfaceType) const;
 
+    /**
+     * @brief Test if this variable permits access through the @p interfaceType.
+     *
+     * Test if this variable permits access through the @p interfaceType. The results
+     * will be given according to this truth table:
+     * 
+     *    argument : stored interface type = return value
+     *    (anything) : public_and_private = true
+     *    private : public = false
+     *    public : private = false
+     *    public, private, public_and_private : none = false
+     *    none : (anything) = true
```

Comment:
This truth table doesn't match the one in https://github.com/cellml/libcellml/pull/718#issue-492718408. No only the "columns" are inverted, but for a `public` stored interface type, to have a `private` argument gives true in one case and false in the other...?!

So, before reviewing further, it would be good to know what is supposed to be correct. Also, I would reformat your truth table. I find it unnecessarily complicated to read and use. Why not simply something like:

```
                   | none | public | private | public_and_private   <--- Parameter
-------------------+------+--------+---------+-------------------
              none |      |        |         |
-------------------+------+--------+---------+-------------------
            public |      |        |         |
-------------------+------+--------+---------+-------------------
           private |      |        |         |
-------------------+------+--------+---------+-------------------
public_and_private |      |        |         |

        ^
        |
 Stored interface
       type
```

Reply:
> Why not simply something like

Because they are a PITA to create ;). OK, I'll check it.

Reply:
> but for a public stored interface type, to have a private argument gives true in one case and false in the other...?!
I can't see that anywhere?

Reply:
a01c97b5

Reply:
I meant that in https://github.com/cellml/libcellml/pull/718#issue-492718408, you originally had:
```
mInterfaceType : parameter -> returns
public : private, public_and_private -> true
```

while from your code above
```
     *    argument : stored interface type = return value
     *    private : public = false
```

Reply:
Yep, found it and changed comment above. Thanks.